### PR TITLE
Implement data gathering functionality of agent

### PR DESCRIPTION
**Pull request by charlieegan3**
Fixes https://github.com/jetstack/preflight/issues/90

### CODE REVIEW

**Comment by charlieegan3** [comment]
Waiting to copy the changes implemented in https://github.com/jetstack/preflight/pull/82

**Comment by j-fuentes** [comment]
I am continuing with this now that #100 has been merged.

**Comment by charlieegan3** [comment]
Thanks Jose :+1:

File: pkg/agent/config_data_gatherers.go [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+	var dg datagatherer.DataGatherer
+
+	if config["data-path"] == "" {
+		return dg, fmt.Errorf("data-path is required")
+	}
+
+	return local.NewLocalDataGatherer(config["data-path"]), nil
+}
+
+// TODO copied from check, perhaps move to utils or leave here until check has gone from the client?
```
Comment (wwwil) [inline pkg/agent/config_data_gatherers.go]: https://github.com/jetstack/preflight/pull/102

Comment (j-fuentes) [reply pkg/agent/config_data_gatherers.go]: That was a leftover. It is completely gone now.

**Comment by j-fuentes** [comment]
@charlieegan3 I have updated this, making use of the new constructors for the data-gatherers.

**Comment by charlieegan3** [comment]
These changes look good to me. 

/approve

**Pull request merged**